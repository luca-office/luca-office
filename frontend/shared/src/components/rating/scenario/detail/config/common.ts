export const WAIT_INTERVAL = 250
