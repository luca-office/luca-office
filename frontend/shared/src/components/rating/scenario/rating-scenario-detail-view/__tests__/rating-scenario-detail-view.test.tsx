import {MockedProvider} from "@apollo/client/testing"
import {act} from "@testing-library/react-hooks"
import {shallow} from "enzyme"
import * as React from "react"
import {create} from "react-test-renderer"
import wait from "waait"
import {fakeTranslate} from "../../../../../../tests/utils/translate-mock"
import {IconName, RaterMode, RatingActionOption} from "../../../../../enums"
import {
  automatedCodingCriteriaMock,
  checkLoginMock,
  codingCriteriaMock,
  codingDimensionsMock,
  ratingsMock,
  scenarioAutomatedCodingItemRatingsMock,
  scenarioManualCodingItemRatingsMock,
  scenariosMock,
  surveyIdMock,
  surveyInvitationIdMock,
  userAccountsMock
} from "../../../../../graphql/__mocks__"
import {QuestionScoringType} from "../../../../../graphql/generated/globalTypes"
import {checkLoginQuery, ratingsQuery, surveyUserAccountsQuery} from "../../../../../graphql/queries"
import {CodingItem} from "../../../../../models"
import {Option} from "../../../../../utils"
import {Card, CardContent} from "../../../../card"
import * as useScenarioCodingItemsByRatingsListHook from "../../../../rating/hooks/use-scenario-coding-items-by-ratings-list"
import {RatingDetailView, RatingOverviewTable} from "../../../common"
import {UseScenarioCodingItemRatingsByRatingsListHook, UseScenarioCodingItemsByRatingsListHook} from "../../../hooks"
import * as useScenarioRatingsByRatingsListHook from "../../../hooks/use-scenario-coding-item-ratings-by-ratings-list"
import {ScenarioAutomaticRatingTable, ScenarioManualRatingTable} from "../../detail"
import * as useRatingScenarioDetailViewHook from "../hooks/use-rating-scenario-detail-view"
import {UseRatingScenarioDetailViewHook} from "../hooks/use-rating-scenario-detail-view"
import {RatingScenarioDetailView, RatingScenarioDetailViewProps} from "../rating-scenario-detail-view"

const scenario = scenariosMock[0]
const codingDimension = codingDimensionsMock[0]
const codingItem = codingDimension.items[0]

const defaultProps: RatingScenarioDetailViewProps = {
  navigateToNextParticipant: jest.fn(),
  navigateToPreviousParticipant: jest.fn(),
  participantIndex: 0,
  participantName: "Laurence Moulder",
  participantsCount: 12,
  navigateToNextQuestion: jest.fn(),
  navigateToPreviousQuestion: jest.fn(),
  surveyInvitationId: surveyInvitationIdMock,
  scenario,
  selectedCodingEntityId: codingItem.id,
  navigateToEntity: jest.fn(),
  ratingId: Option.of(ratingsMock[0].id),
  mode: RaterMode.FinalRater,
  surveyId: surveyIdMock,
  participantFinishedModule: true,
  isReadonly: false,
  isNotRatable: false,
  refreshData: jest.fn()
}

const stateHookValuesDefault: UseRatingScenarioDetailViewHook = {
  dataLoading: false,
  selectedCodingDimension: Option.none(),
  selectedCodingItem: Option.of<CodingItem>(codingItem),
  label: codingItem.title,
  description: codingItem.description,
  isOverviewPage: false,
  isAutomatedCodingItem: false,
  score: 12,
  maxScore: 24,
  backgroundIcon: Option.of<IconName>(IconName.Gear),
  overviewEntityName: "",
  overviewEntities: [],
  scenarioCodingItemRating: Option.of(scenarioManualCodingItemRatingsMock[0]),
  codingCriteria: codingCriteriaMock.map(mock => ({...mock, itemId: codingItem.id})),
  automatedCodingCriteria: automatedCodingCriteriaMock.map(mock => ({...mock, itemId: codingItem.id})),
  scoringType: codingItem.scoringType as unknown as QuestionScoringType,
  selectedRatingAction: RatingActionOption.None,
  setSelectedRatingAction: jest.fn(),
  refreshData: jest.fn(),
  createScenarioCodingItemRatingLoading: false,
  isRatingInProgress: true,
  averageScore: 3
}

const stateSpy = jest.spyOn(useRatingScenarioDetailViewHook, "useRatingScenarioDetailView")

const scenarioRatingsByRatingsListHookValuesDefault: UseScenarioCodingItemRatingsByRatingsListHook = {
  scenarioCodingItemRatings: scenarioAutomatedCodingItemRatingsMock,
  scenarioCodingItemRatingsLoading: false,
  getScenarioCodingItemRatings: jest.fn()
}

const scenarioRatingsByRatingsListSpy = jest.spyOn(
  useScenarioRatingsByRatingsListHook,
  "useScenarioCodingItemRatingsByRatingsList"
)

const scenarioCodingItemsByRatingsListHookValuesDefault: UseScenarioCodingItemsByRatingsListHook = {
  scenarioCodingItemRatingsLoading: false,
  scenarioCodingItemRatings: scenarioManualCodingItemRatingsMock,
  getScenarioCodingItemRatings: jest.fn(() => Promise.resolve(scenarioManualCodingItemRatingsMock))
}
const scenarioCodingItemsByRatingsListSpy = jest.spyOn(
  useScenarioCodingItemsByRatingsListHook,
  "useScenarioCodingItemsByRatingsList"
)

const getComponent = (props?: Partial<RatingScenarioDetailViewProps>) => (
  <MockedProvider
    mocks={[
      {
        request: {
          query: checkLoginQuery
        },
        result: {
          data: {
            checkLogin: checkLoginMock
          }
        }
      },
      {
        request: {
          query: ratingsQuery,
          variables: {surveyId: surveyIdMock}
        },
        result: {
          data: {
            ratings: ratingsMock
          }
        }
      },
      {
        request: {
          query: surveyUserAccountsQuery,
          variables: {surveyId: surveyIdMock}
        },
        result: {
          data: {
            userAccountsForSurvey: userAccountsMock
          }
        }
      }
    ]}>
    <RatingScenarioDetailView {...{...defaultProps, ...props}} />
  </MockedProvider>
)

describe("rating-scenario-detail-view", () => {
  it("renders correctly (selected item)", async () => {
    stateSpy.mockReturnValue(stateHookValuesDefault)
    scenarioRatingsByRatingsListSpy.mockReturnValue(scenarioRatingsByRatingsListHookValuesDefault)
    scenarioCodingItemsByRatingsListSpy.mockReturnValue(scenarioCodingItemsByRatingsListHookValuesDefault)
    const component = create(getComponent())
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
  it("renders correctly (selected dimension)", async () => {
    stateSpy.mockReturnValue({
      ...stateHookValuesDefault,
      selectedCodingDimension: Option.of(codingDimension),
      label: codingDimension.title,
      description: codingDimension.description,
      isOverviewPage: true,
      isAutomatedCodingItem: false,
      overviewEntityName: `${fakeTranslate("rating_scenario__items_label")} (${codingDimension.items.length})`,
      overviewEntities: [
        {
          position: 0,
          title: codingItem.title,
          score: 6,
          maxScore: 12,
          id: codingItem.id,
          rated: true,
          averageScore: 3
        }
      ],
      scoringType: QuestionScoringType.None
    })
    scenarioRatingsByRatingsListSpy.mockReturnValue(scenarioRatingsByRatingsListHookValuesDefault)
    scenarioCodingItemsByRatingsListSpy.mockReturnValue(scenarioCodingItemsByRatingsListHookValuesDefault)
    const component = create(getComponent({selectedCodingEntityId: codingItem.id}))
    await act(() => wait(0))
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
  it("renders correctly (no selection)", async () => {
    stateSpy.mockReturnValue({
      ...stateHookValuesDefault,
      selectedCodingDimension: Option.none(),
      selectedCodingItem: Option.none(),
      label: scenario.name,
      description: scenario.description,
      isOverviewPage: true,
      isAutomatedCodingItem: false,
      scoringType: QuestionScoringType.None
    })
    scenarioRatingsByRatingsListSpy.mockReturnValue(scenarioRatingsByRatingsListHookValuesDefault)
    scenarioCodingItemsByRatingsListSpy.mockReturnValue(scenarioCodingItemsByRatingsListHookValuesDefault)
    const component = create(getComponent({selectedCodingEntityId: undefined}))
    await act(() => wait(0))
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
  it("has correct structure (selected item, isAutomatedCodingItem=false)", async () => {
    stateSpy.mockReturnValue(stateHookValuesDefault)
    scenarioRatingsByRatingsListSpy.mockReturnValue(scenarioRatingsByRatingsListHookValuesDefault)
    scenarioCodingItemsByRatingsListSpy.mockReturnValue(scenarioCodingItemsByRatingsListHookValuesDefault)
    const tree = shallow(getComponent()).childAt(0).dive()

    await act(() => wait(0))

    const ratingDetailView = tree.find(RatingDetailView)
    expect(ratingDetailView).toHaveLength(1)

    const card = ratingDetailView.dive().find(Card)
    expect(card).toHaveLength(1)

    const cardContent = card.dive().find(CardContent)
    expect(cardContent).toHaveLength(1)

    const ratingContent = cardContent.dive().find(".rating-content")
    expect(ratingContent).toHaveLength(1)

    expect(ratingContent.find(RatingOverviewTable)).toHaveLength(0)
    expect(ratingContent.find(ScenarioManualRatingTable)).toHaveLength(1)
    expect(ratingContent.find(ScenarioAutomaticRatingTable)).toHaveLength(0)
  })
  it("has correct structure (selected item, isAutomatedCodingItem=true)", async () => {
    stateSpy.mockReturnValue({...stateHookValuesDefault, isAutomatedCodingItem: true})
    scenarioRatingsByRatingsListSpy.mockReturnValue(scenarioRatingsByRatingsListHookValuesDefault)
    scenarioCodingItemsByRatingsListSpy.mockReturnValue(scenarioCodingItemsByRatingsListHookValuesDefault)
    const tree = shallow(getComponent()).childAt(0).dive()

    await act(() => wait(0))

    const ratingDetailView = tree.find(RatingDetailView)
    expect(ratingDetailView).toHaveLength(1)

    const card = ratingDetailView.dive().find(Card)
    expect(card).toHaveLength(1)

    const cardContent = card.dive().find(CardContent)
    expect(cardContent).toHaveLength(1)

    const ratingContent = cardContent.dive().find(".rating-content")
    expect(ratingContent).toHaveLength(1)

    expect(ratingContent.find(RatingOverviewTable)).toHaveLength(0)
    expect(ratingContent.find(ScenarioManualRatingTable)).toHaveLength(0)
    expect(ratingContent.find(ScenarioAutomaticRatingTable)).toHaveLength(1)
  })
  it("has correct structure (selected dimension)", async () => {
    stateSpy.mockReturnValue({
      ...stateHookValuesDefault,
      selectedCodingDimension: Option.of(codingDimension),
      label: codingDimension.title,
      description: codingDimension.description,
      isOverviewPage: true,
      isAutomatedCodingItem: false,
      overviewEntityName: `${fakeTranslate("rating_scenario__items_label")} (${codingDimension.items.length})`,
      overviewEntities: [
        {
          position: 0,
          title: codingItem.title,
          score: 6,
          maxScore: 12,
          id: codingItem.id,
          rated: true,
          averageScore: 3
        }
      ],
      scoringType: QuestionScoringType.None
    })
    scenarioRatingsByRatingsListSpy.mockReturnValue(scenarioRatingsByRatingsListHookValuesDefault)
    scenarioCodingItemsByRatingsListSpy.mockReturnValue(scenarioCodingItemsByRatingsListHookValuesDefault)
    const tree = shallow(getComponent({selectedCodingEntityId: codingItem.id}))
      .childAt(0)
      .dive()

    await act(() => wait(0))

    const ratingDetailView = tree.find(RatingDetailView)
    expect(ratingDetailView).toHaveLength(1)

    const card = ratingDetailView.dive().find(Card)
    expect(card).toHaveLength(1)

    const cardContent = card.dive().find(CardContent)
    expect(cardContent).toHaveLength(1)

    const ratingContent = cardContent.dive().find(".rating-content")
    expect(ratingContent).toHaveLength(1)

    expect(ratingContent.find(RatingOverviewTable)).toHaveLength(1)
    expect(ratingContent.find(ScenarioManualRatingTable)).toHaveLength(0)
    expect(ratingContent.find(ScenarioAutomaticRatingTable)).toHaveLength(0)
  })
  it("has correct structure (no selection)", async () => {
    stateSpy.mockReturnValue({
      ...stateHookValuesDefault,
      selectedCodingDimension: Option.none(),
      selectedCodingItem: Option.none(),
      label: scenario.name,
      description: scenario.description,
      isOverviewPage: true,
      isAutomatedCodingItem: false,
      scoringType: QuestionScoringType.None
    })
    scenarioRatingsByRatingsListSpy.mockReturnValue(scenarioRatingsByRatingsListHookValuesDefault)
    scenarioCodingItemsByRatingsListSpy.mockReturnValue(scenarioCodingItemsByRatingsListHookValuesDefault)
    const tree = shallow(getComponent({selectedCodingEntityId: undefined}))
      .childAt(0)
      .dive()

    await act(() => wait(0))

    const ratingDetailView = tree.find(RatingDetailView)
    expect(ratingDetailView).toHaveLength(1)

    const card = ratingDetailView.dive().find(Card)
    expect(card).toHaveLength(1)

    const cardContent = card.dive().find(CardContent)
    expect(cardContent).toHaveLength(1)

    const ratingContent = cardContent.dive().find(".rating-content")
    expect(ratingContent).toHaveLength(1)

    expect(ratingContent.find(RatingOverviewTable)).toHaveLength(1)
    expect(ratingContent.find(ScenarioManualRatingTable)).toHaveLength(0)
    expect(ratingContent.find(ScenarioAutomaticRatingTable)).toHaveLength(0)
  })
})
