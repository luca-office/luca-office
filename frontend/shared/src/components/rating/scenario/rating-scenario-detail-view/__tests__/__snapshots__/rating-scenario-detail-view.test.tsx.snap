// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rating-scenario-detail-view renders correctly (no selection) 1`] = `
<div
  className="css-16c40fq-Card"
>
  <div
    className="card-header css-k3h7c1-CardHeader"
  >
    <div
      className="css-1phnlf7-Icon"
      dangerouslySetInnerHTML={
        {
          "__html": {},
        }
      }
    />
    <div
      className="css-17zbm7y-Text"
    >
      rating__labeled_count
    </div>
    <div
      className="css-1ut7gm9-RatingDetailHeader"
    >
      <button
        className="css-1cywa3f-Button"
        disabled={false}
        onClick={[Function]}
      >
        <div
          className="css-95j2hm-Button"
        >
          <div
            className="css-mj8vo1-Icon"
            dangerouslySetInnerHTML={
              {
                "__html": {},
              }
            }
          />
        </div>
      </button>
      <button
        className="css-1cywa3f-Button"
        disabled={false}
        onClick={[Function]}
      >
        <div
          className="css-95j2hm-Button"
        >
          <div
            className="css-mj8vo1-Icon"
            dangerouslySetInnerHTML={
              {
                "__html": {},
              }
            }
          />
        </div>
      </button>
    </div>
  </div>
  <div
    className="card-content css-1kceaqr-CardContent"
  >
    <div
      className="css-dmq2pf-Icon"
      dangerouslySetInnerHTML={
        {
          "__html": {},
        }
      }
    />
    <div
      className="css-afaif4-RatingDetailView"
    >
      <div
        className="css-1d2tndp-Heading"
      >
        Scenario 1
      </div>
      <div
        className="css-1e4yjc9-Text"
      >
        description
      </div>
      <div
        className="rating-content css-z5r6dl-RatingDetailView"
      >
        <div
          className="css-0"
        >
          <div
            className="css-1i94808-Heading"
          >
            
          </div>
          <div
            className="css-0"
          >
            <div
              className="css-giha9y-Table"
            >
              <div
                className="css-1kpl4zp-Table"
              >
                <div
                  className="css-fuywjr-Table"
                >
                  <div
                    className="css-1ahn5qo-RatingOverviewTable"
                  >
                    <div
                      className="css-1sbf56p-Text"
                    >
                      title
                    </div>
                  </div>
                </div>
                <div
                  className="css-105qn2y-Table"
                >
                  <div
                    className="css-1sbf56p-Text"
                  >
                    rating_scenario__scoring_label
                  </div>
                </div>
              </div>
              <div
                className="css-tqgzls-Table"
              >
                <div
                  className="table-placeholder css-1v6q0ml-Table"
                />
              </div>
            </div>
            <div
              className="card-footer css-jb8qoz-CardFooter"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="card-footer css-1uma7zq-CardFooter"
  >
    <div
      className="css-1ck0okw-RatingDetailFooter"
    >
      <div
        className="css-icd26n-RatingDetailFooter"
      >
        <div
          className="css-17zbm7y-Text"
        >
          rating__rating__rating_global_score
        </div>
        <div
          className="css-3c95an-RatingDetailFooter"
        >
          <div
            className="css-dtunoo-Text"
          >
            rating__rating__scoring
          </div>
          <div
            className="css-1nfczpt-Icon"
            dangerouslySetInnerHTML={
              {
                "__html": {},
              }
            }
          />
        </div>
      </div>
      <div
        className="css-ujroht-ProgressBar"
      >
        <div
          className="progress-element css-zcdupw-ProgressBar"
        />
      </div>
    </div>
    <div
      className="css-1dsw8ki-RatingDetailFooter"
    >
      <button
        className="css-1cywa3f-Button"
        disabled={false}
        onClick={[Function]}
      >
        <div
          className="css-95j2hm-Button"
        >
          <div
            className="css-mj8vo1-Icon"
            dangerouslySetInnerHTML={
              {
                "__html": {},
              }
            }
          />
        </div>
      </button>
      <button
        className="css-1cywa3f-Button"
        disabled={false}
        onClick={[Function]}
      >
        <div
          className="css-95j2hm-Button"
        >
          <div
            className="css-mj8vo1-Icon"
            dangerouslySetInnerHTML={
              {
                "__html": {},
              }
            }
          />
        </div>
      </button>
    </div>
  </div>
</div>
`;

exports[`rating-scenario-detail-view renders correctly (selected dimension) 1`] = `
<div
  className="css-16c40fq-Card"
>
  <div
    className="card-header css-k3h7c1-CardHeader"
  >
    <div
      className="css-1phnlf7-Icon"
      dangerouslySetInnerHTML={
        {
          "__html": {},
        }
      }
    />
    <div
      className="css-17zbm7y-Text"
    >
      rating__labeled_count
    </div>
    <div
      className="css-1ut7gm9-RatingDetailHeader"
    >
      <button
        className="css-1cywa3f-Button"
        disabled={false}
        onClick={[Function]}
      >
        <div
          className="css-95j2hm-Button"
        >
          <div
            className="css-mj8vo1-Icon"
            dangerouslySetInnerHTML={
              {
                "__html": {},
              }
            }
          />
        </div>
      </button>
      <button
        className="css-1cywa3f-Button"
        disabled={false}
        onClick={[Function]}
      >
        <div
          className="css-95j2hm-Button"
        >
          <div
            className="css-mj8vo1-Icon"
            dangerouslySetInnerHTML={
              {
                "__html": {},
              }
            }
          />
        </div>
      </button>
    </div>
  </div>
  <div
    className="card-content css-1kceaqr-CardContent"
  >
    <div
      className="css-dmq2pf-Icon"
      dangerouslySetInnerHTML={
        {
          "__html": {},
        }
      }
    />
    <div
      className="css-afaif4-RatingDetailView"
    >
      <div
        className="css-1d2tndp-Heading"
      >
        Argumentation (inhaltlich)
      </div>
      <div
        className="css-1e4yjc9-Text"
      >
        dsfsdf
      </div>
      <div
        className="rating-content css-z5r6dl-RatingDetailView"
      >
        <div
          className="css-0"
        >
          <div
            className="css-1i94808-Heading"
          >
            rating_scenario__items_label (1)
          </div>
          <div
            className="css-0"
          >
            <div
              className="css-giha9y-Table"
            >
              <div
                className="css-1kpl4zp-Table"
              >
                <div
                  className="css-fuywjr-Table"
                >
                  <div
                    className="css-1ahn5qo-RatingOverviewTable"
                  >
                    <div
                      className="css-1sbf56p-Text"
                    >
                      title
                    </div>
                  </div>
                </div>
                <div
                  className="css-105qn2y-Table"
                >
                  <div
                    className="css-1sbf56p-Text"
                  >
                    rating_scenario__scoring_label
                  </div>
                </div>
              </div>
              <div
                className="css-tqgzls-Table"
              >
                <div
                  className="css-1ihrgr1-Table"
                >
                  <div
                    className="entity-row css-22vfow-Table"
                    onClick={[Function]}
                  >
                    <div
                      className="css-fuywjr-Table"
                    >
                      <div
                        className="css-20cfnl-Table"
                      >
                        <div
                          className="css-1sbf56p-Text"
                        >
                          1. Item 1
                        </div>
                      </div>
                    </div>
                    <div
                      className="css-105qn2y-Table"
                    >
                      <div
                        className="css-1w7lpcq-Table"
                      >
                        <div
                          className="css-198wifw-Text"
                        >
                          rating__rating__scoring
                        </div>
                        <div
                          className="css-1p4ztl7-Icon"
                          dangerouslySetInnerHTML={
                            {
                              "__html": {},
                            }
                          }
                        />
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              className="card-footer css-jb8qoz-CardFooter"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="card-footer css-1uma7zq-CardFooter"
  >
    <div
      className="css-1ck0okw-RatingDetailFooter"
    >
      <div
        className="css-icd26n-RatingDetailFooter"
      >
        <div
          className="css-17zbm7y-Text"
        >
          rating__rating__rating_global_score
        </div>
        <div
          className="css-3c95an-RatingDetailFooter"
        >
          <div
            className="css-dtunoo-Text"
          >
            rating__rating__scoring
          </div>
          <div
            className="css-1nfczpt-Icon"
            dangerouslySetInnerHTML={
              {
                "__html": {},
              }
            }
          />
        </div>
      </div>
      <div
        className="css-ujroht-ProgressBar"
      >
        <div
          className="progress-element css-zcdupw-ProgressBar"
        />
      </div>
    </div>
    <div
      className="css-1dsw8ki-RatingDetailFooter"
    >
      <button
        className="css-1cywa3f-Button"
        disabled={false}
        onClick={[Function]}
      >
        <div
          className="css-95j2hm-Button"
        >
          <div
            className="css-mj8vo1-Icon"
            dangerouslySetInnerHTML={
              {
                "__html": {},
              }
            }
          />
        </div>
      </button>
      <button
        className="css-1cywa3f-Button"
        disabled={false}
        onClick={[Function]}
      >
        <div
          className="css-95j2hm-Button"
        >
          <div
            className="css-mj8vo1-Icon"
            dangerouslySetInnerHTML={
              {
                "__html": {},
              }
            }
          />
        </div>
      </button>
    </div>
  </div>
</div>
`;

exports[`rating-scenario-detail-view renders correctly (selected item) 1`] = `
<div
  className="css-16c40fq-Card"
>
  <div
    className="card-header css-k3h7c1-CardHeader"
  >
    <div
      className="css-1phnlf7-Icon"
      dangerouslySetInnerHTML={
        {
          "__html": {},
        }
      }
    />
    <div
      className="css-17zbm7y-Text"
    >
      rating__labeled_count
    </div>
    <div
      className="css-1ut7gm9-RatingDetailHeader"
    >
      <button
        className="css-1cywa3f-Button"
        disabled={false}
        onClick={[Function]}
      >
        <div
          className="css-95j2hm-Button"
        >
          <div
            className="css-mj8vo1-Icon"
            dangerouslySetInnerHTML={
              {
                "__html": {},
              }
            }
          />
        </div>
      </button>
      <button
        className="css-1cywa3f-Button"
        disabled={false}
        onClick={[Function]}
      >
        <div
          className="css-95j2hm-Button"
        >
          <div
            className="css-mj8vo1-Icon"
            dangerouslySetInnerHTML={
              {
                "__html": {},
              }
            }
          />
        </div>
      </button>
    </div>
  </div>
  <div
    className="card-content css-1kceaqr-CardContent"
  >
    <div
      className="css-dmq2pf-Icon"
      dangerouslySetInnerHTML={
        {
          "__html": {},
        }
      }
    />
    <div
      className="css-afaif4-RatingDetailView"
    >
      <div
        className="css-1d2tndp-Heading"
      >
        Item 1
      </div>
      <div
        className="css-1e4yjc9-Text"
      >
        Item 1
      </div>
      <div
        className="rating-content css-z5r6dl-RatingDetailView"
      >
        <div
          className="css-fdluvi-CriterionRatingTable"
        >
          <div
            className="css-9r06kc-CriterionRatingTable"
          >
            <div
              className="css-75nvf3-CriterionRatingTable"
            >
              <div
                className="css-x84t8t-Heading"
              >
                rating__manual_rating_label (5)
              </div>
              <div
                className="css-0"
                id="tooltip_42"
              >
                <div
                  className="css-qnyrc1-Icon"
                  dangerouslySetInnerHTML={
                    {
                      "__html": {},
                    }
                  }
                />
              </div>
              <div
                className="react-tooltip styles-module_tooltip__mnnfp styles-module_dark__xNqje css-10zyuev-Tooltip"
                role="tooltip"
                style={
                  {
                    "backgroundColor": "#2e3032",
                    "zIndex": 1,
                  }
                }
              >
                <div
                  className="css-1tdk34p-Tooltip"
                >
                  <div
                    className="css-1kjzr28-Tooltip"
                  >
                    <div
                      className="css-jb7kz7-Heading"
                    >
                      rating__manual_rating_label
                    </div>
                    <div
                      className="css-8tih5m-Text"
                    >
                      rating__manual_rating_description
                    </div>
                  </div>
                  <div
                    className="css-1fvvs0r-Tooltip"
                  >
                    <div
                      className="css-nlq2eo-Icon"
                      dangerouslySetInnerHTML={
                        {
                          "__html": {},
                        }
                      }
                    />
                  </div>
                </div>
                <div
                  className="react-tooltip-arrow styles-module_arrow__K0L3T"
                  style={{}}
                />
              </div>
            </div>
            <div>
              <div
                className="css-18lu61u-Table"
              >
                <div
                  className="css-1kpl4zp-Table"
                >
                  <div
                    className="css-fuywjr-Table"
                  >
                    <div
                      className="css-jwlj9x-Text"
                    >
                      rating__criterion_label
                    </div>
                  </div>
                  <div
                    className="css-181zh2l-Table"
                  >
                    <div
                      className="css-0"
                      id="tooltip_42"
                    >
                      <div
                        className="css-1j062c4-CriterionRatingTable"
                      >
                        <div
                          className="css-jwlj9x-Text"
                        >
                          rating__scoring_analytic_title
                        </div>
                        <div
                          className="css-l0jg45-Icon"
                          dangerouslySetInnerHTML={
                            {
                              "__html": {},
                            }
                          }
                        />
                      </div>
                    </div>
                    <div
                      className="react-tooltip styles-module_tooltip__mnnfp styles-module_dark__xNqje css-10zyuev-Tooltip"
                      role="tooltip"
                      style={
                        {
                          "backgroundColor": "#2e3032",
                          "zIndex": 1,
                        }
                      }
                    >
                      <div
                        className="css-1tdk34p-Tooltip"
                      >
                        <div
                          className="css-1kjzr28-Tooltip"
                        >
                          <div
                            className="css-jb7kz7-Heading"
                          >
                            rating__scoring_analytic_title
                          </div>
                          <div
                            className="css-8tih5m-Text"
                          >
                            coding_item_update__selection_card_type_analytical_description
                          </div>
                        </div>
                        <div
                          className="css-1fvvs0r-Tooltip"
                        >
                          <div
                            className="css-nlq2eo-Icon"
                            dangerouslySetInnerHTML={
                              {
                                "__html": {},
                              }
                            }
                          />
                        </div>
                      </div>
                      <div
                        className="react-tooltip-arrow styles-module_arrow__K0L3T"
                        style={{}}
                      />
                    </div>
                  </div>
                </div>
                <div
                  className="css-tqgzls-Table"
                >
                  <div
                    className="css-1e2w3lh-ContentLoadingIndicator"
                  >
                    <div
                      className="css-1tzcb72-Icon"
                      dangerouslySetInnerHTML={
                        {
                          "__html": {},
                        }
                      }
                    />
                  </div>
                </div>
              </div>
              <div
                className="card-footer css-1nm6ew4-CardFooter"
              >
                <div
                  className="css-dccfv8-CriterionRatingTable"
                >
                  <div
                    className="css-1eg3ylp-CriterionRatingTable"
                  >
                    <div
                      className="css-1udb1oe-CriterionRatingTable"
                    >
                      <div
                        className="css-1kh99ct-CriterionRatingTable"
                      >
                        <div
                          className="css-x84t8t-Heading"
                        >
                          rating__no_criterion_title
                        </div>
                      </div>
                      <div
                        className="css-1969l7h-CriterionRatingTable"
                      >
                        <button
                          className="css-1kbijgd-Button"
                          disabled={false}
                          onClick={[Function]}
                        >
                          <div
                            className="css-95j2hm-Button"
                          >
                            rating__no_score_button
                          </div>
                        </button>
                      </div>
                    </div>
                  </div>
                  <div
                    className="css-1orwcfe-CriterionRatingTable"
                  >
                    <div
                      className="css-hluplg-RatingDetailActionElement"
                    >
                      <div
                        className="css-nzx43c-RatingDetailActionElement"
                      >
                        <div
                          className="css-143noxf-RatingDetailActionElement"
                        >
                          <div
                            className="css-14rj767"
                            data-testid="custom-select"
                          >
                            <div
                              className="css-rarr6p"
                            >
                              <div
                                className="css-13s2sfk-Heading"
                              >
                                rating__rating__action_dropdown_label
                              </div>
                              <div
                                className="css-0"
                                id="tooltip_42"
                              >
                                <div
                                  className="css-qnyrc1-Icon"
                                  dangerouslySetInnerHTML={
                                    {
                                      "__html": {},
                                    }
                                  }
                                />
                              </div>
                              <div
                                className="react-tooltip styles-module_tooltip__mnnfp styles-module_dark__xNqje css-10zyuev-Tooltip"
                                role="tooltip"
                                style={
                                  {
                                    "backgroundColor": "#2e3032",
                                    "zIndex": 1,
                                  }
                                }
                              >
                                <div
                                  className="css-1tdk34p-Tooltip"
                                >
                                  <div
                                    className="css-1kjzr28-Tooltip"
                                  >
                                    <div
                                      className="css-jb7kz7-Heading"
                                    >
                                      rating__rating__action_dropdown_label
                                    </div>
                                    <div
                                      className="css-8tih5m-Text"
                                    >
                                      rating__rating__action_dropdown_tooltip_text
                                    </div>
                                  </div>
                                  <div
                                    className="css-1fvvs0r-Tooltip"
                                  >
                                    <div
                                      className="css-nlq2eo-Icon"
                                      dangerouslySetInnerHTML={
                                        {
                                          "__html": {},
                                        }
                                      }
                                    />
                                  </div>
                                </div>
                                <div
                                  className="react-tooltip-arrow styles-module_arrow__K0L3T"
                                  style={{}}
                                />
                              </div>
                            </div>
                            <div
                              className="css-18vooac"
                              onMouseEnter={[Function]}
                              onMouseLeave={[Function]}
                            >
                              <div
                                className="custom-select"
                              >
                                MockedSelect
                              </div>
                            </div>
                          </div>
                        </div>
                        <div
                          className="css-e24v2t-RatingDetailActionElement"
                        >
                          <div
                            className="css-0"
                            id="tooltip_42"
                          >
                            <button
                              className="css-injsqh-Button"
                              disabled={true}
                              onClick={[Function]}
                            >
                              <div
                                className="css-95j2hm-Button"
                              >
                                <div
                                  className="css-1xq4w0l-Icon"
                                  dangerouslySetInnerHTML={
                                    {
                                      "__html": {},
                                    }
                                  }
                                />
                                confirmed
                              </div>
                            </button>
                          </div>
                          <div
                            className="react-tooltip styles-module_tooltip__mnnfp styles-module_dark__xNqje css-10zyuev-Tooltip"
                            role="tooltip"
                            style={
                              {
                                "backgroundColor": "#2e3032",
                                "zIndex": 1,
                              }
                            }
                          >
                            rating__action_button_not_possible
                            <div
                              className="react-tooltip-arrow styles-module_arrow__K0L3T"
                              style={{}}
                            />
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="card-footer css-1uma7zq-CardFooter"
  >
    <div
      className="css-1ck0okw-RatingDetailFooter"
    >
      <div
        className="css-icd26n-RatingDetailFooter"
      >
        <div
          className="css-17zbm7y-Text"
        >
          rating__rating__rating_score
        </div>
        <div
          className="css-3c95an-RatingDetailFooter"
        >
          <div
            className="css-dtunoo-Text"
          >
            rating__rating__scoring
          </div>
          <div
            className="css-1nfczpt-Icon"
            dangerouslySetInnerHTML={
              {
                "__html": {},
              }
            }
          />
        </div>
      </div>
      <div
        className="css-ujroht-ProgressBar"
      >
        <div
          className="progress-element css-zcdupw-ProgressBar"
        />
      </div>
    </div>
    <div
      className="css-1dsw8ki-RatingDetailFooter"
    >
      <button
        className="css-1cywa3f-Button"
        disabled={false}
        onClick={[Function]}
      >
        <div
          className="css-95j2hm-Button"
        >
          <div
            className="css-mj8vo1-Icon"
            dangerouslySetInnerHTML={
              {
                "__html": {},
              }
            }
          />
        </div>
      </button>
      <button
        className="css-1cywa3f-Button"
        disabled={false}
        onClick={[Function]}
      >
        <div
          className="css-95j2hm-Button"
        >
          <div
            className="css-mj8vo1-Icon"
            dangerouslySetInnerHTML={
              {
                "__html": {},
              }
            }
          />
        </div>
      </button>
    </div>
  </div>
</div>
`;
